src/_types.ts

```diff
@@ -21,20 +21,15 @@ export type IterableContainer<T = unknown> = ReadonlyArray<T> | readonly [];
 
 // Inspired and largely copied from `sindresorhus/ts-extras`:
 // @see https://github.com/sindresorhus/ts-extras/blob/44f57392c5f027268330771996c4fdf9260b22d6/source/object-keys.ts
-export type ObjectKeys<T extends object> = `${Exclude<keyof T, symbol>}`;
+export type EnumeratedKeyOf<T> = `${Exclude<keyof T, symbol>}`;
```

Comment:
This is a nit, but you might consider renaming `EnumeratedKeyOf` (and `EnumeratedValueOf` below) to something like `NonSymbolKeyOf`. The current name makes it sound like it has to do with enumerable vs non-enumerable keys, which is orthogonal to string vs symbol keys.

Reply:
Renamed to EnumerableStringKeyOf, following the definition provided in: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#description

Reply:
Still a nit, but I think `Enumerable` in the name is still misleading. It could make sense to name a *function* `enumerableStringKeysOf` (because it describes what the function does as runtime), but this is a type definition, and the type system doesn't distinguish between enumerable and non-enumerable keys.

For example, the builtin type definition for `Error` includes the `message` property. If you call `EnumerableStringKeyOf<Error>`, you'll get back a type that still includes `message`. But if you call `Object.keys(new Error())`, `message` won't be included, because it isn't enumerable.

Reply:
Handling the JS built-ins this way is outside the scope of what we try to do with typing in Remeda. In general, we don't test classes in most of our utils; the issue with classes (and interface declaration merging, getter/setter functions, etc...) is that they complicate the typing system considerably, and the main reason that Typescript doesn't provide the sort of types we aim to provide for the built-in functions.
Most people avoid using these features, which allows us to simplify our types. If you already have non-enumerable string keys in an object, it's an advanced use case for which something other than general-purpose helper utils might be a better fit.

Even considering symbols in our types is more advanced than what the vast majority of our users need, and that played into how I named this type; adding "symbols" to the name brings in an advanced concept to a type that would be used by users who have no dealing with symbols in their code, and would force them to consider what this means. Referring to enumerability in the name is a hint to users who do understand what that means that this function has caveats when dealing with these concepts, without negatively impacting the experience for those who don't.

TL;DR - Naming is hard...